pkg/util/taints/taints.go

```diff
@@ -159,22 +160,23 @@ func deleteTaints(taintsToRemove []v1.Taint, newTaints *[]v1.Taint) ([]error, bo
 	return allErrs, removed
 }
 
-// addTaints adds the newTaints list to existing ones and updates the newTaints List.
-// TODO: This needs a rewrite to take only the new values instead of appended newTaints list to be consistent.
-func addTaints(oldTaints []v1.Taint, newTaints *[]v1.Taint) bool {
+// addedTaints returns the taints that exists in oldTaints but not in newTaints.
+// It returns true when len(oldTains) != (len(*newTaints) + len(addedTaints)).
+// If there are taints in the newTaints but not the oldTaints, return nil and true.
+func addedTaints(oldTaints []v1.Taint, newTaints []v1.Taint) (addedTaints []v1.Taint, lenChanged bool) {
```

Comment:
Please declare variables rather than using named returns - they increase reading complexity here (especially with the name dupe), without giving any functional advantages.

Reply:
You're right, thank you!

I have updated and squash commits.